init/bootstrap.go

```diff
@@ -43,10 +46,18 @@ outer:
 			continue
 		}
 
-		for _, b := range buffer {
-			if b != 0 {
-				log.Infof("%s not empty", dev)
-				continue outer
+		boot2docker := false
+
+		if strings.HasPrefix(string(buffer), boot2dockerMagic) {
```

Comment:
Interesting, maybe string([]byte) does do a copy.  Now I'm going to dig into that a bit further.